scripts/irods/test/test_ils.py

```diff
@@ -57,3 +57,20 @@ def test_genquery_fallback_for_collection_permissions__issue_4636(self):
                           " R_OBJT_ACCESS.access_type_id = R_TOKN_MAIN.token_id")
         self.admin.assert_icommand(['iadmin', 'asq', specific_query, 'ShowCollAcls'])
 
+    def test_ils_lists_collections_without_listing_their_contents__issue_5506(self):
```

Comment:
need a test to show that `-A` still works?

Reply:
Should they be incompatible? Right now, they are ignored if the arg is a collection. Same for the -r option.

Reply:
i've come around....

for collection
 - `-d` and `-l` are fine - `-d` takes precendence  - ignore `-l`
 - `-d` and `-L` are fine - `-d` takes precendence  - ignore `-L`

for data object
 - `-r` is ignored?   matches behavior of `ls -R` (recursive)
 - `-d` should be... ignored as well... same as (`ls -d foo.txt`)

Reply:
Great! I believe the code modifications support all of your thoughts right now. The -d option is only checked on the collection listing code path.

I'll add tests so that this behavior is captured.

Reply:
done.